prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -204,12 +235,25 @@ def get_function_bullet_point_list(model: str) -> str:
         "To start, click 'Run' to see the agent in action, or bootstrap the "
         "agent with your own reasoning."
     )
-    st.markdown(
-        f"It is equipped with the following tools to access the "
-        f"[{MARKET_TYPE}]({MARKET_TYPE.market_class.base_url}) prediction "
-        f"market APIs:"
-    )
-    st.markdown(get_function_bullet_point_list(model=model))
+    st.markdown("It is equipped with the following tools:")
+    if agent_is_initialized():
+        st.markdown(
+            get_function_bullet_point_list(agent=st.session_state.agent, model=model)
+        )
+    else:
+        st.markdown("The agent is not initialized yet.")
+
+with st.expander("Agent's current system prompt"):
```

Comment:
I haven't done a very good job of it in this app.py, but the intention is to have a separation between layout (the top part, where everything that's not static is just a container) and execution (the bottom part under `# Execution`).

The idea being that we don't always want things to run in the same order that they are laid out in. And also it should be easy-ish from just reading the code what the layout will look like, and what order things will run in. These new bits are dynamic, so I guess there should be a placeholder here and the markdown should be generated further down

Reply:
@gabrielfior noticed that as it currently is, the prompt is updated in the UI with a 1-iteration lag, because it gets written before the agent is run (and potentially does calls 'update system prompt').

I know you're looking at this in https://github.com/gnosis/prediction-market-agent/pull/238. What do you think about this suggested approach?

Reply:
I will create a separate PR for solving this 1-iteration lag (spoiler alert - it involves adding callbacks to streamlit elements like buttons).
Regarding separating `Execution` and `Layout`, also fine and fits nicely with my changes from above.